### PR TITLE
Bug 1858528 - Add rakuten and yahoo-jp-auctions search engines

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1858528

### CODE REVIEW

**Comment by sarah541** [comment]
@Standard8 I added these search engines based on desktop search extensions as mentioned for these two search engines i.e [rakuten](https://searchfox.org/mozilla-central/source/browser/components/search/extensions/rakuten/manifest.json) and [yahoo-jp-auctions](https://searchfox.org/mozilla-central/source/browser/components/search/extensions/yahoo-jp-auctions/manifest.json). I converted the favicon.ico for both engines to base64-encoded PNG image and added it to the xml.

**Review by rocketsroger — APPROVED** [review]
🚢 Thanks!

**Review by Standard8 — CHANGES_REQUESTED** [review]
Thank you, this looks good, just a couple of minor corrections before we're ready to land.

File: android-components/components/feature/search/src/main/assets/searchplugins/rakuten.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <Url method="GET" template="https://pt.afl.rakuten.co.jp/c/013ca98b.cd7c5f0c/" type="text/html">
+        <Param name="v" value="2" />
```
[the remark this answers is not in the document]
Comment (sarah541) [reply android-components/components/feature/search/src/main/assets/searchplugins/rakuten.xml]: Eagle eye! Thank you Mark

File: android-components/components/feature/search/src/main/assets/searchplugins/rakuten.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+
+<SearchPlugin xmlns="http://www.mozilla.org/2006/browser/search/">
+    <ShortName>楽天市場</ShortName>
+    <Description>楽天市場 商品検索</Description>
+    <InputEncoding>utf-8</InputEncoding>
```
[the remark this answers is not in the document]
Comment (sarah541) [reply android-components/components/feature/search/src/main/assets/searchplugins/rakuten.xml]: That's fair! I have updated the Pr with both comments

**Review by Standard8 — APPROVED** [review]
Thank you for the updates.

**Pull request merged**